boltdb.go

```diff
@@ -62,60 +62,66 @@ func NewBoltDBWithOpts(name string, dir string, opts *bbolt.Options) (DB, error)
 	return &BoltDB{db: db}, nil
 }
 
-func (bdb *BoltDB) Get(key []byte) (value []byte) {
+func (bdb *BoltDB) Get(key []byte) (value []byte, err error) {
 	key = nonEmptyKey(nonNilBytes(key))
-	err := bdb.db.View(func(tx *bbolt.Tx) error {
+	err = bdb.db.View(func(tx *bbolt.Tx) error {
 		b := tx.Bucket(bucket)
 		if v := b.Get(key); v != nil {
 			value = append([]byte{}, v...)
 		}
 		return nil
 	})
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 	return
 }
 
 func (bdb *BoltDB) Has(key []byte) bool {
-	return bdb.Get(key) != nil
+	bytes, err := bdb.Get(key)
+	if err != nil {
+		return false
+	}
+	return bytes != nil
 }
 
-func (bdb *BoltDB) Set(key, value []byte) {
+func (bdb *BoltDB) Set(key, value []byte) error {
 	key = nonEmptyKey(nonNilBytes(key))
 	value = nonNilBytes(value)
 	err := bdb.db.Update(func(tx *bbolt.Tx) error {
 		b := tx.Bucket(bucket)
 		return b.Put(key, value)
 	})
 	if err != nil {
-		panic(err)
+		return err
 	}
+	return nil
 }
 
-func (bdb *BoltDB) SetSync(key, value []byte) {
-	bdb.Set(key, value)
+func (bdb *BoltDB) SetSync(key, value []byte) error {
```

Comment:
What is the purpose of this function?

Reply:
in boltdbs case, yes but in other dbs (goleveldb) it does have different functionality. so to adhere to the interface it is there. We can speak about re working the interface, but not sure people will be up for that

Reply:
Thank you!